src/modeling_library/dist/dist_dsl.jl

```diff
@@ -145,24 +143,20 @@ end
 macro dist(fnexpr)
   # First, pull out arguments and body
   fndef = splitdef(longdef(fnexpr))
-  #
-  # if !@capture(fnexpr, argexpr_ -> body_)
-  #     error("@dist expression must be of the form @dist (args...) -> body")
-  # end
-  # arguments = (typeof(argexpr) == Symbol) ? (argexpr,) : argexpr.args # [] #argexpr
-
   arguments = fndef[:args]
   body = fndef[:body]
 
   name_to_index = Dict{Symbol, Int8}()
+  name_to_type = Dict{Symbol, Type}()
   for (i, arg) in enumerate(arguments)
-      (argname, _, _, _) = splitarg(arg)
+      (argname, argtype, _, _) = splitarg(arg)
       name_to_index[argname] = i
+      name_to_type[argname] = eval(argtype)
```

Comment:
I just realized that one potential issue here is that this may perform the `eval` with respect to Gen, and not the calling module...? And so user-defined types may be ignored...?

Reply:
Yeah -- since you're just embedding this type into generated code anyway, why not just use `argtype` directly (rather than `eval(argtype)`?

Reply:
Just changed this line to `__module__.eval(argtype)`, which works as expected! I think the reason I can't use `argtype` directly is because `SimpleArg`s are constructed within the `@dist` macro, and so I need to provide the type parameter then and there.